src/engine/SpatialJoin.cpp

```diff
@@ -309,10 +355,13 @@ PreparedSpatialJoinParams SpatialJoin::prepareJoin() const {
     return std::pair{idTablePtr, std::move(resTable)};
   };
 
-  auto getJoinCol = [](const std::shared_ptr<const QueryExecutionTree>& child,
+  auto getVarColMap =
+      [](const std::shared_ptr<const QueryExecutionTree>& child) {
+        return child->getRootOperation()->getExternallyVisibleVariableColumns();
+      };
+
+  auto getJoinCol = [](VariableToColumnMap varColMap,
                        const Variable& childVariable) {
-    auto varColMap =
-        child->getRootOperation()->getExternallyVisibleVariableColumns();
     return varColMap[childVariable].columnIndex_;
```

Comment:
I think this would increase the complexity to read this code because there are multiple operations on this VarToColMap which can currently be cleanly handled by `varToCol[var]` and `varToCol.contains(var)`. Do you agree?

Reply:
As commented below I confused this review with another location in the code. Resolving this now.